aws_jobs/cyhy-data-extract.py

```diff
@@ -132,40 +134,36 @@ def query_data(collection, query, tbz_file, tbz_filename, end_of_data_collection
         end_of_data_collection.isoformat().replace(":", "").split(".")[0],
     )
     collection_file = open(json_filename, "w")
-    # How many documents into a query that will be skipped
-    skips = 0
-    # Number of documents in a query
-    count = collection.find(query, {"key": False}).count()
-    while skips < count:
-        # Pull documents between n and n + 100000
-        collection_file.write(
-            to_json(
-                list(
-                    collection.find(query, {"key": False}).skip(skips).limit(PAGE_SIZE)
-                )
-            )
-        )
-        skips += PAGE_SIZE
+    collection_file.write("[")
+
+    last_id = None
+    while True:
+        if last_id is None:
+            result = collection.find(
+                query, {"key": False}, sort=[("_id", pymongo.ASCENDING)]
+            ).limit(PAGE_SIZE)
+        else:
+            query["_id"] = {"$gt": last_id}
+            result = collection.find(
+                query, {"key": False}, sort=[("_id", pymongo.ASCENDING)]
+            ).limit(PAGE_SIZE)
+
+        for doc in result:
+            collection_file.write(to_json([doc])[1:-2])
+            collection_file.write(",")
+
+        last_id = doc["_id"]
+
+        if result.retrieved == 0:
+            # If we have outputted documents then we have a trailing comma, so we
+            # need to roll back the file location to overwrite as we finish
```

Comment:
Small cleanup request:

\# If we **~~have~~** output**~~ted~~** documents then we have a trailing comma, so we
\# need to roll back the file location **by one byte** to overwrite as we finish